eos/scattering/ee-to-ccbar.cc

```diff
@@ -0,0 +1,673 @@
+            const double GeVtonb = 10 * pow( 1.e18 * hbar * speedoflight, 2.0);
+
+            // Channel properties
+            const double Nf = 2 * K->_channels[channel]->_l_orbital + 1;
+            const double rhof = real(K->_channels[channel]->rho(intermediate_result->s));
+
+            // Get T-matrix[ee, channel]
+            const complex<double> T1f = intermediate_result->tmatrix_row_0[channel];
+
+            return sqrt(GeVtonb * 16. * M_PI / intermediate_result->s * Nf * rhof) * T1f;
+        }
+
+        double sigma_eetochannel(const IntermediateResult * intermediate_result, const unsigned & channel)
+        {
+            return norm(amplitude_eetochannel(intermediate_result, channel));
+        }
+
+        // Widths
+        double res_partial_width(unsigned res, unsigned channel)
+        {
+            return K->partial_width(res, channel);
+        }
+
+        double res_total_width(unsigned res)
+        {
+            return K->width(res);
+        }
+
+        // Ratios
+        double R(const IntermediateResult * intermediate_result)
+        {
+            double total_xsec = 0.0;
+
+            for (unsigned i = 1; i < nchannels; i++)
+            {
+                total_xsec += sigma_eetochannel(intermediate_result, i);
+            }
+
+            return total_xsec / sigma_eetomumu(intermediate_result->E) + Rconstant; //Add constant term
+        }
+
+        double Rc(const IntermediateResult * intermediate_result)
+        {
+            double total_xsec = 0.0;
+
+            for (unsigned i = 6; i < nchannels; i++)
+            {
+                total_xsec += sigma_eetochannel(intermediate_result, i);
+            }
+
+            return total_xsec / sigma_eetomumu(intermediate_result->E);
+        }
+    };
+
+    const std::vector<std::string>
+    Implementation<EEToCCBar>::resonance_names = { "psi(2S)", "psi(3770)", "psi(4040)", "psi(4160)", "psi(4415)" };
+
+    const std::vector<std::string>
+    Implementation<EEToCCBar>::channel_names = { "ee", "eff(2S)", "eff(3770)", "eff(4040)", "eff(4160)", "eff(4415)",
+            "D^0Dbar^0", "D^+D^-", "D^0Dbar^*0", "D^*0Dbar^0", "D^+D^*-", "D^*+D^-", "D_s^+D_s^-", "D^*0Dbar^*0P0",
+            "D^*0Dbar^*0P2", "D^*0Dbar^*0F2", "D^*+D^*-P0", "D^*+D^*-P2", "D^*+D^*-F2", "D_s^+D_s^*-", "D_s^*+D_s^-",
+            "D_s^*+D_s^*-P0", "D_s^*+D_s^*-P2", "D_s^*+D_s^*-F2" };
+
+
+    EEToCCBar::EEToCCBar(const Parameters & parameters, const Options & options) :
+        PrivateImplementationPattern<EEToCCBar>(new Implementation<EEToCCBar>(parameters, options, *this))
+    {
+    }
+
+    EEToCCBar::~EEToCCBar()
+    {
+    }
+
+    const std::vector<OptionSpecification>
+    Implementation<EEToCCBar>::options
+    {
+    };
+
+    const EEToCCBar::IntermediateResult *
+    EEToCCBar::prepare(const double & E) const
+    {
+        return _imp->prepare(E);
+    }
+
+    double
+    EEToCCBar::psi2S_ee_width() const
+    {
+        return _imp->res_partial_width(0, 0);
```

Comment:
Hm, I'm not so happy with the use of "magic" numbers here.
Can we use an ``enum Resonances = { psi2S = 0, ... };``?
And similar ``enum Channel = { ee = 0, ... };``?

Reply:
I completely agree. However, we need to discuss how to implement this properly. Esp. should it be at the `kmatrix-impl.hh` level or only at the `ee-to-ccbar.cc`?

Reply:
Should be at the level of ``ee-to-ccbar.{cc,hh}``.